site/en/guides/porting/define-cmake-rules.md

```diff
@@ -0,0 +1,135 @@
+# Define CMake rules
+
+The CMake build system relies on `CMakeLists.txt` files which define build targets. The `CMakeLists.txt` at the root of your repository is the top of the build tree and is a good place to start defining the various targets, options, and macros which will be used in the build process.
+
+The very first values to define are the project name and the supported platforms.
+
+In this example, we set the project name to `ot-efr32` with a version of `0.0.1`. We also define a variable `EFR32_PLATFORM_VALUES` which is a list of `efr32` platforms supported by `ot-efr32`. For the sake of this example, we've defined multiple platforms, but having a single platform for the `_PLATFORM_VALUES` variable is fine as well.
+
+```cmake
+cmake_minimum_required(VERSION 3.10.2)
+project(ot-efr32 VERSION 0.0.1)
+
+set(EFR32_PLATFORM_VALUES
+    "efr32mg1"
+    "efr32mg12"
+    "efr32mg13"
+    "efr32mg21"
+)
+```
+
+The `CMakeLists.txt` file includes a check that will prevent a build launched for an unsupported platform.
+```cmake
+set_property(CACHE EFR32_PLATFORM PROPERTY STRINGS ${EFR32_PLATFORM_VALUES})
```

Comment:
Yup! A dollar sign followed by a term in curly braces is how you get the value of a variable in CMake. See more [here](https://cmake.org/cmake/help/v3.0/manual/cmake-language.7.html#variable-references)

**Example**
```cmake
# Define a variable named "MY_VARIABLE" with a value of "foobar"
set(MY_VARIABLE foobar)

# Print value of MY_VARIABLE and exit
message(FATAL_ERROR "The value of MY_VARIABLE is ${MY_VARIABLE}")
```

^^ This would print
```
The value of MY_VARIABLE is foobar
```

Reply:
Thanks, I'll be able to handle it properly in the import script.